compiler/include/type.h

```diff
@@ -113,6 +113,70 @@ class Type : public BaseAST {
 
 #define forv_Type(_p, _v) forv_Vec(Type, _p, _v)
 
+
+// a Qualifier allows the compiler to distinguish between
+// different properties of a variable (const or ref-ness in particular)
+// without changing its type to a ref or wide ref type.
+typedef enum {
```

Comment:
I am surprised that this is typedef enum { } Qualifier; rather than just enum Qualifier{ };